presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -227,6 +234,12 @@ public TaskStatus process()
                     return TaskStatus.finished();
                 }
                 ListenableFuture<?> future;
+                if (dynamicFilteringProbeBlockingTimeoutMillis.get() > 0 && !dynamicFilter.isComplete()) {
```

Comment:
add `long startTime` and timeout after `dynamicFilteringProbeBlockingTimeoutMillis - startTime`?

Reply:
Changed to use a StopWatch